pkg/provisioner/crdprovisioner.go

```diff
@@ -278,6 +283,12 @@ func (c *CrdProvisioner) DeleteVolume(ctx context.Context, volumeID string, secr
 		return err
 	}
 
+	// if volume deletion already in process, return to prevent duplicate request
+	if azVolume.Status.State == v1alpha1.VolumeDeleting {
+		return status.Errorf(codes.Internal, "deletion still in process for volume (%s)", volumeName)
```

Comment:
`codes.Aborted`?

Reply:
Ditto for remaning cases where `codes.Internal` is returned for a duplicate/pending op.